impl/src/main/java/org/jboss/weld/bootstrap/events/ContainerLifecycleEvents.java

```diff
@@ -241,6 +252,9 @@ public void fireProcessBean(BeanManagerImpl beanManager, Bean<?> bean) {
             } else if (bean instanceof ProducerMethod<?, ?>) {
                 ProcessProducerMethodImpl.fire(beanManager, (ProducerMethod<?, ?>) bean);
             } else {
+                if (isProcessSynthethicBeanObserved()) {
```

Comment:
I wouldn't detect `processSynthethicBeanObserved` at all (we don't detect `ProcessSyntheticAnnotatedType` neither). Instead, just check the extension param:

    if (extension != null) {